### PR TITLE
API/UCP: align ucp_worker_fence description with oshmem

### PR DESCRIPTION
**Pull request by hoopoepg**
- align description of ucp_worker_fence with shmem_fence

### CODE REVIEW

**Comment by brminich** [comment]
LGTM,
@yosefe, @shamisp

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -3811,10 +3811,10 @@ ucs_status_t ucp_dt_query(ucp_datatype_t datatype, ucp_datatype_attr_t *attr);
...
  *
  * This routine ensures ordering of non-blocking communication operations on
- * the @ref ucp_worker_h "UCP worker".  Communication operations issued on the
- * @a worker prior to this call are guaranteed to be completed before any
- * subsequent communication operations to the same @ref ucp_worker_h "worker"
- * which follow the call to @ref ucp_worker_fence "fence".
+ * the @ref ucp_worker_h "UCP worker". Communication operations issued to a
+ * particular endpoint created on the @a worker prior to this call are
+ * guaranteed to be completed before any subsequent communication operations to
+ * the same endpoint which follow the call to @ref ucp_worker_fence "fence".
```
Comment (yosefe) [inline src/ucp/api/ucp.h:3817]:
```
Communication operations issued on a particular endpoint created on the @a worker 
prior to this call are guaranteed to be completed before any communication operations 
issued on the same endpoint after this call.
```

Comment (hoopoepg) [reply src/ucp/api/ucp.h]: updated

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by yosefe** [comment]
@tonycurtis @shamisp can you pls review?

**Review by tonycurtis — APPROVED** [review]

**Review by shamisp — APPROVED** [review]

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**